File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    result = [0]

    for i in numbers:
        tmp = []
        for j in result:
            tmp.append(j + i)
            tmp.append(j - i)
        result = tmp

    for num in result:
        if num == target:
            answer += 1

    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

int answer = 0;

void DFS(vector<int> numbers, int target, int pos, int sum){
    if(pos == numbers.size()){
        if(sum == target){
            answer++;
        }
        return;
    }
    
    DFS(numbers, target, pos+1, sum + numbers[pos]);
    DFS(numbers, target, pos+1, sum - numbers[pos]);
}

int solution(vector<int> numbers, int target) {
    DFS(numbers, target, 0, 0);
    return answer;
}